src/schnorrsig.rs

```diff
@@ -142,9 +141,13 @@ impl KeyPair {
 
     /// Creates a Schnorr KeyPair directly from generic Secp256k1 secret key
     ///
-    /// Panics if internal representation of the provided [`SecretKey`] does not
-    /// holds correct secret key value obtained from Secp256k1 library
-    /// previously
+    /// # Panic
+    ///
+    /// Panics if internal representation of the provided [`SecretKey`] does not hold correct secret
+    /// key value obtained from Secp256k1 library previously, for instance when the corresponding
+    /// public key is not even key.
+    // TODO: Consider returning `Result`, since even if the secret key is constructed with Secp256k1
+    //       library, the function still will panic on keys producing odd public key values.
```

Comment:
This doesn't sound right. AFAIK if the resultant public key were to be odd, it will simply be negated.

Reply:
No, it's even simpler than that, the `keypair` type simply has no restrictions on key parity.